src/solid/Virtualizer.tsx

```diff
@@ -280,9 +284,36 @@ export const Virtualizer = <T,>(props: VirtualizerProps<T>): JSX.Element => {
     })
   );
 
-  const dataSlice = createMemo<T[]>(() => {
+  const dataSlice = createMemo<{ item: T, index: number }[]>(() => {
     const [start, end] = range();
-    return end >= 0 ? props.data.slice(start, end + 1) : [];
+    const items: { item: T, index: number }[] = [];
+    
+    for (let i = start, j = end; i <= j; i++) {
+        const item = props.data[i];
+        if (item === undefined) continue;
+        
+        items.push({ item, index: i });
+    }
+
+    if (props.keepMounted) {
+        const startItems: typeof items = [];
+        const endItems: typeof items = [];
+
+        for (const idx of props.keepMounted.toSorted()) {
+            if (idx >= start && idx <= end) continue;
+            
+            const item = props.data[idx];
+            if (item === undefined) continue;
+
+            const list = idx < start ? startItems : endItems;
+            list.push({ item, index: idx });
```

Comment:
should I instead declare that it is not undefined?

```suggestion
            const item = props.data[idx]!;

            const list = idx < start ? startItems : endItems;
            list.push({ item, index: idx });
```